### ISSUE TITLE
Missmatch in core dimension

### ISSUE THREAD

**Issue by jo-mueller** [BodenmillerGroup/napari-czifile2#4]
Hi!

I'm getting the following error after loading czi files in napar with czifile2:

``` ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 5 is different from 6)```

I can't really make anything from this...

Here is the full traceback:

````
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\napari\_qt\qt_viewer.py in _open_files_dialog(self=<napari._qt.qt_viewer.QtViewer object>)
    638
    639         if (filenames != []) and (filenames is not None):
--> 640             self.viewer.open(filenames)
        self.viewer.open = <bound method ViewerModel.open of Viewer(axes=Axes(visible=False, labels=True, colored=True, dashed=False, arrows=True), camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, interactive=True), cursor=Cursor(position=(1.0, 1.0), scaled=True, size=1, style='standard'), dims=Dims(ndim=2, ndisplay=2, last_used=0, range=((0, 2, 1), (0, 2, 1)), current_step=(0, 0), order=(0, 1), axis_labels=('0', '1')), grid=GridCanvas(enabled=False, stride=1, shape=(-1, -1)), layers=[], scale_bar=ScaleBar(visible=False, colored=False, ticks=True, position='bottom_right', font_size=10.0, unit=None), text_overlay=TextOverlay(visible=False, color=array([1., 1., 1., 1.]), font_size=10.0, position='top_left', text=''), help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_move_callbacks=[], mouse_drag_callbacks=[], mouse_wheel_callbacks=[<function dims_scroll at 0x000001DE33AE6B80>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, keymap={'Control-Shift-C': <bound method QtViewer.toggle_console_visibility of <napari._qt.qt_viewer.QtViewer object at 0x000001DE3402D700>>})>
        filenames = ['E:/Promotion/Projects/2020_Radiomics/Data/N182e_SAS_27/Histology/1_181211/181212_N182e_SAS_27_1_IF.czi']
    641             update_open_history(filenames[0])
    642

c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\napari\components\viewer_model.py in open(self=Viewer(axes=Axes(visible=False, labels=True, col..._viewer.QtViewer object at 0x000001DE3402D700>>}), path=['E:/Promotion/Projects/2020_Radiomics/Data/N182e_...7/Histology/1_181211/181212_N182e_SAS_27_1_IF.czi'], stack=False, plugin=None, layer_type=None, **kwargs={})
    854         for _path in paths:
    855             added.extend(
--> 856                 self._add_layers_with_plugins(
        self._add_layers_with_plugins = <bound method ViewerModel._add_layers_with_plugins of Viewer(axes=Axes(visible=False, labels=True, colored=True, dashed=False, arrows=True), camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, interactive=True), cursor=Cursor(position=(1.0, 1.0), scaled=True, size=1, style='standard'), dims=Dims(ndim=2, ndisplay=2, last_used=0, range=((0, 2, 1), (0, 2, 1)), current_step=(0, 0), order=(0, 1), axis_labels=('0', '1')), grid=GridCanvas(enabled=False, stride=1, shape=(-1, -1)), layers=[], scale_bar=ScaleBar(visible=False, colored=False, ticks=True, position='bottom_right', font_size=10.0, unit=None), text_overlay=TextOverlay(visible=False, color=array([1., 1., 1., 1.]), font_size=10.0, position='top_left', text=''), help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_move_callbacks=[], mouse_drag_callbacks=[], mouse_wheel_callbacks=[<function dims_scroll at 0x000001DE33AE6B80>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, keymap={'Control-Shift-C': <bound method QtViewer.toggle_console_visibility of <napari._qt.qt_viewer.QtViewer object at 0x000001DE3402D700>>})>
        _path = 'E:/Promotion/Projects/2020_Radiomics/Data/N182e_SAS_27/Histology/1_181211/181212_N182e_SAS_27_1_IF.czi'
        kwargs = {}
        plugin = None
        layer_type = None
    857                     _path, kwargs, plugin=plugin, layer_type=layer_type
    858                 )

c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\napari\components\viewer_model.py in _add_layers_with_plugins(self=Viewer(axes=Axes(visible=False, labels=True, col..._viewer.QtViewer object at 0x000001DE3402D700>>}), path_or_paths='E:/Promotion/Projects/2020_Radiomics/Data/N182e_...7/Histology/1_181211/181212_N182e_SAS_27_1_IF.czi', kwargs={}, plugin='napari-czifile2', layer_type=None)
    927             # actually add the layer
    928             with layer_source(path=filename, reader_plugin=plugin):
--> 929                 added.extend(self._add_layer_from_data(*_data))
        added.extend = <built-in method extend of list object at 0x000001DE42D6CE00>
        self._add_layer_from_data = <bound method ViewerModel._add_layer_from_data of Viewer(axes=Axes(visible=False, labels=True, colored=True, dashed=False, arrows=True), camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, interactive=True), cursor=Cursor(position=(1.0, 1.0), scaled=True, size=1, style='standard'), dims=Dims(ndim=2, ndisplay=2, last_used=0, range=((0, 2, 1), (0, 2, 1)), current_step=(0, 0), order=(0, 1), axis_labels=('0', '1')), grid=GridCanvas(enabled=False, stride=1, shape=(-1, -1)), layers=[], scale_bar=ScaleBar(visible=False, colored=False, ticks=True, position='bottom_right', font_size=10.0, unit=None), text_overlay=TextOverlay(visible=False, color=array([1., 1., 1., 1.]), font_size=10.0, position='top_left', text=''), help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_move_callbacks=[], mouse_drag_callbacks=[], mouse_wheel_callbacks=[<function dims_scroll at 0x000001DE33AE6B80>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, keymap={'Control-Shift-C': <bound method QtViewer.toggle_console_visibility of <napari._qt.qt_viewer.QtViewer object at 0x000001DE3402D700>>})>
        _data = (array((many many numbers here), dtype=uint16), {'rgb': False, 'channel_axis': 2, 'translate': (0.0, 0.0, 0.0, 0.0), 'scale': (1.0, 1.0, 0.645, 0.645), 'name': ['Rhodamine', 'EGFP', 'DAPI']}, 'image')
    930         return added
    931

c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\napari\components\viewer_model.py in _add_layer_from_data(self=Viewer(axes=Axes(visible=False, labels=True, col..._viewer.QtViewer object at 0x000001DE3402D700>>}), data=array([[[[[[  0],
           [  0],
           [...      [458],
           [458]]]]]], dtype=uint16), meta={'channel_axis': 2, 'name': ['Rhodamine', 'EGFP', 'DAPI'], 'rgb': False, 'scale': (1.0, 1.0, 0.645, 0.645), 'translate': (0.0, 0.0, 0.0, 0.0)}, layer_type='image')
   1001         try:
   1002             add_method = getattr(self, 'add_' + layer_type)
-> 1003             layer = add_method(data, **(meta or {}))
        layer = undefined
        add_method = <bound method ViewerModel.add_image of Viewer(axes=Axes(visible=False, labels=True, colored=True, dashed=False, arrows=True), camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, interactive=True), cursor=Cursor(position=(1.0, 1.0), scaled=True, size=1, style='standard'), dims=Dims(ndim=2, ndisplay=2, last_used=0, range=((0, 2, 1), (0, 2, 1)), current_step=(0, 0), order=(0, 1), axis_labels=('0', '1')), grid=GridCanvas(enabled=False, stride=1, shape=(-1, -1)), layers=[], scale_bar=ScaleBar(visible=False, colored=False, ticks=True, position='bottom_right', font_size=10.0, unit=None), text_overlay=TextOverlay(visible=False, color=array([1., 1., 1., 1.]), font_size=10.0, position='top_left', text=''), help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_move_callbacks=[], mouse_drag_callbacks=[], mouse_wheel_callbacks=[<function dims_scroll at 0x000001DE33AE6B80>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, keymap={'Control-Shift-C': <bound method QtViewer.toggle_console_visibility of <napari._qt.qt_viewer.QtViewer object at 0x000001DE3402D700>>})>
        data = array([(many many numbers here)
        meta = {'rgb': False, 'channel_axis': 2, 'translate': (0.0, 0.0, 0.0, 0.0), 'scale': (1.0, 1.0, 0.645, 0.645), 'name': ['Rhodamine', 'EGFP', 'DAPI']}
   1004         except TypeError as exc:
   1005             if 'unexpected keyword argument' not in str(exc):

c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\napari\components\viewer_model.py in add_image(self=Viewer(axes=Axes(visible=False, labels=True, col..._viewer.QtViewer object at 0x000001DE3402D700>>}), data=array([[[[[[  0],
           [  0],
           [...      [458],
           [458]]]]]], dtype=uint16), channel_axis=2, rgb=False, colormap=None, contrast_limits=None, gamma=1, interpolation='nearest', rendering='mip', iso_threshold=0.5, attenuation=0.05, name=['Rhodamine', 'EGFP', 'DAPI'], metadata=None, scale=(1.0, 1.0, 0.645, 0.645), translate=(0.0, 0.0, 0.0, 0.0), rotate=None, shear=None, affine=None, opacity=1, blending=None, visible=True, multiscale=None)
    701             layer_list = list()
    702             for image, i_kwargs, _ in layerdata_list:
--> 703                 layer = Image(image, **i_kwargs)
        layer = undefined
        global Image = <class 'napari.layers.image.image.Image'>
        image = (many many numbers here)
        i_kwargs = {'rgb': False, 'colormap': 'cyan', 'contrast_limits': None, 'gamma': 1, 'interpolation': 'nearest', 'rendering': 'mip', 'iso_threshold': 0.5, 'attenuation': 0.05, 'name': 'Rhodamine', 'metadata': None, 'scale': (1.0, 1.0, 0.645, 0.645), 'translate': (0.0, 0.0, 0.0, 0.0), 'rotate': None, 'shear': None, 'affine': None, 'opacity': 1, 'blending': 'additive', 'visible': True, 'multiscale': False}
    704                 self.layers.append(layer)
    705                 layer_list.append(layer)

c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\napari\layers\image\image.py in __init__(self=<Image layer 'Rhodamine'>, data=array([[[[[  0],
          [  0],
          [  0...        [252],
          [252]]]]], dtype=uint16), rgb=False, colormap='cyan', contrast_limits=None, gamma=1, interpolation='nearest', rendering='mip', iso_threshold=0.5, attenuation=0.05, name='Rhodamine', metadata=None, scale=(1.0, 1.0, 0.645, 0.645), translate=(0.0, 0.0, 0.0, 0.0), rotate=None, shear=None, affine=None, opacity=1, blending='additive', visible=True, multiscale=False)
    288
    289         # Trigger generation of view slice and thumbnail
--> 290         self._update_dims()
        self._update_dims = <bound method Layer._update_dims of <Image layer 'Rhodamine' at 0x1de40062640>>
    291
    292     def _new_empty_slice(self):

c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\napari\layers\base\base.py in _update_dims(self=<Image layer 'Rhodamine'>, event=None)
    542         self._ndim = ndim
    543
--> 544         self.refresh()
        self.refresh = <bound method Layer.refresh of <Image layer 'Rhodamine' at 0x1de40062640>>
    545
    546     @property

c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\napari\layers\base\base.py in refresh(self=<Image layer 'Rhodamine'>, event=None)
    944         """Refresh all layer data based on current view slice."""
    945         if self.visible:
--> 946             self.set_view_slice()
        self.set_view_slice = <bound method Layer.set_view_slice of <Image layer 'Rhodamine' at 0x1de40062640>>
    947             self.events.set_data()
    948             self._update_thumbnail()

c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\napari\layers\base\base.py in set_view_slice(self=<Image layer 'Rhodamine'>)
    808     def set_view_slice(self):
    809         with self.dask_optimized_slicing():
--> 810             self._set_view_slice()
        self._set_view_slice = <bound method _ImageBase._set_view_slice of <Image layer 'Rhodamine' at 0x1de40062640>>
    811
    812     @abstractmethod

c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\napari\layers\image\image.py in _set_view_slice(self=<Image layer 'Rhodamine'>)
    517
    518         # Check if requested slice outside of data range
--> 519         indices = np.array(self._slice_indices)
        global indices = undefined
        global np.array = <built-in function array>
        self._slice_indices = undefined
    520         extent = self._extent_data
    521         if np.any(

c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\napari\layers\base\base.py in _slice_indices(self=<Image layer 'Rhodamine'>)
    610     def _slice_indices(self):
    611         """(D, ) array: Slice indices in data coordinates."""
--> 612         inv_transform = self._data_to_world.inverse
        inv_transform = undefined
        self._data_to_world.inverse = undefined
    613
    614         if self.ndim > self._ndisplay:

c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\napari\layers\base\base.py in _data_to_world(self=<Image layer 'Rhodamine'>)
    994         affine * (rotate * shear * scale + translate)
    995         """
--> 996         return self._transforms[1:3].simplified
        self._transforms.simplified = undefined
    997
    998     def _update_draw(self, scale_factor, corner_pixels, shape_threshold):

c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\napari\utils\transforms\transforms.py in simplified(self=[<napari.utils.transforms.transforms.Affine obje....transforms.Affine object at 0x000001DE40062850>])
    123             return self[0]
    124         else:
--> 125             return tz.pipe(self[0], *[tf.compose for tf in self[1:]])
        global tz.pipe = <function pipe at 0x000001DE301D3820>
        self = [<napari.utils.transforms.transforms.Affine object at 0x000001DE40062910>, <napari.utils.transforms.transforms.Affine object at 0x000001DE40062850>]
        global tf.compose = undefined
        global tf = undefined
    126
    127     def set_slice(self, axes: Sequence[int]) -> 'TransformChain':

c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\toolz\functoolz.py in pipe(data=<napari.utils.transforms.transforms.Affine object>, *funcs=(<bound method Affine.compose of <napari.utils.transforms.transforms.Affine object>>,))
    625     """
    626     for func in funcs:
--> 627         data = func(data)
        data = <napari.utils.transforms.transforms.Affine object at 0x000001DE40062910>
        func = <bound method Affine.compose of <napari.utils.transforms.transforms.Affine object at 0x000001DE40062850>>
    628     return data
    629

c:\users\johan\anaconda3\envs\napari-env\lib\site-packages\napari\utils\transforms\transforms.py in compose(self=<napari.utils.transforms.transforms.Affine object>, transform=<napari.utils.transforms.transforms.Affine object>)
    458     def compose(self, transform: 'Affine') -> 'Affine':
    459         """Return the composite of this transform and the provided one."""
--> 460         affine_matrix = self.affine_matrix @ transform.affine_matrix
        affine_matrix = undefined
        self.affine_matrix = array([[1., 0., 0., 0., 0., 0.],
       [0., 1., 0., 0., 0., 0.],
       [0., 0., 1., 0., 0., 0.],
       [0., 0., 0., 1., 0., 0.],
       [0., 0., 0., 0., 1., 0.],
       [0., 0., 0., 0., 0., 1.]])
        transform.affine_matrix = array([[1.   , 0.   , 0.   , 0.   , 0.   ],
       [0.   , 1.   , 0.   , 0.   , 0.   ],
       [0.   , 0.   , 0.645, 0.   , 0.   ],
       [0.   , 0.   , 0.   , 0.645, 0.   ],
       [0.   , 0.   , 0.   , 0.   , 1.   ]])
    461         return Affine(affine_matrix=affine_matrix)
    462

ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 5 is different from 6)
```

**Comment by jwindhager** [comment]
Hi @jo-mueller, thanks for reporting this issue. What version of napari-czifile2 are you using? Could you share the output of `pip list` or `conda list`? Would you be able to (privately) share the .czi file with me for testing to jonas.windhager(at)uzh.ch?

**Comment by jo-mueller** [comment]
Hi @jwindhager,

thanks for the quick reply. The version of napari-czifile2 is 0.2.1 (installed it first only today)

Here goes my list of packages:

```
# Name                    Version                   Build  Channel
aicsimageio               3.3.4                    pypi_0    pypi
aicspylibczi              2.8.0                    pypi_0    pypi
alabaster                 0.7.12             pyhd3eb1b0_0
appdirs                   1.4.4                    pypi_0    pypi
argh                      0.26.2                   py38_0
argon2-cffi               20.1.0           py38h2bbff1b_1
arrow                     0.13.1                   py38_0
asciitree                 0.3.3                    pypi_0    pypi
astroid                   2.5.8            py38haa95532_0
async_generator           1.10               pyhd3eb1b0_0
atomicwrites              1.4.0                      py_0
attrs                     21.2.0             pyhd3eb1b0_0
autopep8                  1.5.6              pyhd3eb1b0_0
babel                     2.9.1              pyhd3eb1b0_0
backcall                  0.2.0              pyhd3eb1b0_0
bcrypt                    3.2.0            py38he774522_0
binaryornot               0.4.4              pyhd3eb1b0_1
black                     19.10b0                    py_0
blas                      1.0                         mkl
bleach                    3.3.0              pyhd3eb1b0_0
blosc                     1.21.0               h19a0ad4_0
brotli                    1.0.9                ha925a31_2
brotlipy                  0.7.0           py38h2bbff1b_1003
bzip2                     1.0.8                he774522_0
ca-certificates           2021.5.25            haa95532_1
cachey                    0.2.1                    pypi_0    pypi
certifi                   2021.5.30        py38haa95532_0
cffi                      1.14.5           py38hcd4344a_0
cfitsio                   3.470                he774522_6
chardet                   4.0.0           py38haa95532_1003
charls                    2.2.0                h6c2663c_0
click                     8.0.1              pyhd3eb1b0_0
cloudpickle               1.6.0                      py_0
colorama                  0.4.4              pyhd3eb1b0_0
cookiecutter              1.7.2              pyhd3eb1b0_0
cryptography              3.4.7            py38h71e12ea_0
cycler                    0.10.0                   py38_0
cytoolz                   0.11.0           py38he774522_0
czifile                   2019.7.2                 pypi_0    pypi
dask                      2021.6.2                 pypi_0    pypi
dask-core                 2021.6.0           pyhd3eb1b0_0
decorator                 5.0.9                    pypi_0    pypi
defusedxml                0.7.1              pyhd3eb1b0_0
diff-match-patch          20200713                   py_0
distributed               2021.6.2                 pypi_0    pypi
dnspython                 2.1.0                    pypi_0    pypi
docstring-parser          0.8.1                    pypi_0    pypi
docutils                  0.17.1           py38haa95532_1
elementpath               2.2.3                    pypi_0    pypi
email-validator           1.1.3                    pypi_0    pypi
entrypoints               0.3                      py38_0
fasteners                 0.16.3                   pypi_0    pypi
flake8                    3.9.0              pyhd3eb1b0_0
freetype                  2.10.4               hd328e21_0
freetype-py               2.2.0                    pypi_0    pypi
fsspec                    2021.6.1                 pypi_0    pypi
future                    0.18.2                   py38_1
giflib                    5.2.1                h62dcd97_0
heapdict                  1.0.1                    pypi_0    pypi
icc_rt                    2019.0.0             h0cc432a_1
icu                       58.2                 ha925a31_3
idna                      2.10               pyhd3eb1b0_0
imagecodecs               2021.6.8         py38h5da4933_0
imageio                   2.9.0              pyhd3eb1b0_0
imagesize                 1.2.0              pyhd3eb1b0_0
importlib-metadata        3.10.0           py38haa95532_0
importlib_metadata        3.10.0               hd3eb1b0_0
inflection                0.5.1            py38haa95532_0
intel-openmp              2021.2.0           haa95532_616
intervaltree              3.1.0                      py_0
ipykernel                 5.5.5                    pypi_0    pypi
ipython                   7.24.1                   pypi_0    pypi
ipython_genutils          0.2.0              pyhd3eb1b0_1
isort                     5.8.0              pyhd3eb1b0_0
jedi                      0.18.0                   pypi_0    pypi
jinja2                    3.0.1                    pypi_0    pypi
jinja2-time               0.2.0              pyhd3eb1b0_2
jpeg                      9b                   hb83a4c4_2
jsonschema                3.2.0                      py_2
jupyter_client            6.1.12             pyhd3eb1b0_0
jupyter_core              4.7.1            py38haa95532_0
jupyterlab_pygments       0.1.2                      py_0
keyring                   23.0.1           py38haa95532_0
kiwisolver                1.3.1            py38hd77b12b_0
lazy-object-proxy         1.6.0            py38h2bbff1b_0
lcms2                     2.12                 h83e58a3_0
lerc                      2.2.1                hd77b12b_0
libaec                    1.0.4                h33f27b4_1
libdeflate                1.7                  h2bbff1b_5
libpng                    1.6.37               h2a8f88b_0
libsodium                 1.0.18               h62dcd97_0
libspatialindex           1.9.3                h6c2663c_0
libtiff                   4.2.0                hd0e1b90_0
libzopfli                 1.0.3                ha925a31_0
locket                    0.2.1                    pypi_0    pypi
lxml                      4.6.3                    pypi_0    pypi
lz4-c                     1.9.3                h2bbff1b_0
m2w64-gcc-libgfortran     5.3.0                         6
m2w64-gcc-libs            5.3.0                         7
m2w64-gcc-libs-core       5.3.0                         7
m2w64-gmp                 6.1.0                         2
m2w64-libwinpthread-git   5.0.0.4634.697f757               2
magicgui                  0.2.9                    pypi_0    pypi
markupsafe                2.0.1                    pypi_0    pypi
matplotlib-base           3.3.4            py38h49ac443_0
matplotlib-inline         0.1.2                    pypi_0    pypi
mccabe                    0.6.1                    py38_1
mistune                   0.8.4           py38he774522_1000
mkl                       2021.2.0           haa95532_296
mkl-service               2.3.0            py38h2bbff1b_1
mkl_fft                   1.3.0            py38h277e83a_2
mkl_random                1.2.1            py38hf11a4ad_2
msgpack                   1.0.2                    pypi_0    pypi
msys2-conda-epoch         20160418                      1
mypy_extensions           0.4.3                    py38_0
napari                    0.4.10                   pypi_0    pypi
napari-aicsimageio        0.2.0                    pypi_0    pypi
napari-console            0.0.3                    pypi_0    pypi
napari-czifile2           0.2.1                    pypi_0    pypi
napari-plugin-engine      0.1.9                    pypi_0    pypi
napari-svg                0.1.5                    pypi_0    pypi
nbclient                  0.5.3              pyhd3eb1b0_0
nbconvert                 6.0.7                    py38_0
nbformat                  5.1.3              pyhd3eb1b0_0
nest-asyncio              1.5.1              pyhd3eb1b0_0
networkx                  2.5.1              pyhd3eb1b0_0
notebook                  6.4.0            py38haa95532_0
numcodecs                 0.8.0                    pypi_0    pypi
numpy                     1.21.0                   pypi_0    pypi
numpy-base                1.20.2           py38hc2deb75_0
numpydoc                  1.1.0              pyhd3eb1b0_1
olefile                   0.46                       py_0
ome-types                 0.2.5                    pypi_0    pypi
openjpeg                  2.3.0                h5ec785f_1
openssl                   1.1.1k               h2bbff1b_0
packaging                 20.9               pyhd3eb1b0_0
pandas                    1.2.5                    pypi_0    pypi
pandoc                    2.12                 haa95532_0
pandocfilters             1.4.3            py38haa95532_1
paramiko                  2.7.2                      py_0
parso                     0.8.2                    pypi_0    pypi
partd                     1.2.0              pyhd3eb1b0_0
pathspec                  0.7.0                      py_0
pexpect                   4.8.0              pyhd3eb1b0_3
pickleshare               0.7.5           pyhd3eb1b0_1003
pillow                    8.2.0            py38h4fa10fc_0
pint                      0.17                     pypi_0    pypi
pip                       21.1.2           py38haa95532_0
pluggy                    0.13.1           py38haa95532_0
poyo                      0.5.0              pyhd3eb1b0_0
prometheus_client         0.11.0             pyhd3eb1b0_0
prompt-toolkit            3.0.19                   pypi_0    pypi
psutil                    5.8.0            py38h2bbff1b_1
ptyprocess                0.7.0              pyhd3eb1b0_2
pycodestyle               2.6.0              pyhd3eb1b0_0
pycparser                 2.20                       py_2
pydantic                  1.8.2                    pypi_0    pypi
pydocstyle                6.1.1              pyhd3eb1b0_0
pyflakes                  2.2.0              pyhd3eb1b0_0
pygments                  2.9.0              pyhd3eb1b0_0
pylint                    2.8.3            py38haa95532_1
pyls-black                0.4.6                hd3eb1b0_0
pyls-spyder               0.3.2              pyhd3eb1b0_0
pynacl                    1.4.0            py38h62dcd97_1
pyopengl                  3.1.5                    pypi_0    pypi
pyopenssl                 20.0.1             pyhd3eb1b0_1
pyparsing                 2.4.7              pyhd3eb1b0_0
pyqt                      5.9.2            py38ha925a31_4
pyqt5                     5.15.4                   pypi_0    pypi
pyqt5-qt5                 5.15.2                   pypi_0    pypi
pyqt5-sip                 12.9.0                   pypi_0    pypi
pyrsistent                0.17.3           py38he774522_0
pysocks                   1.7.1            py38haa95532_0
python                    3.8.10               hdbf39b2_7
python-dateutil           2.8.1              pyhd3eb1b0_0
python-jsonrpc-server     0.4.0                      py_0
python-language-server    0.36.2             pyhd3eb1b0_0
python-slugify            5.0.2              pyhd3eb1b0_0
pytz                      2021.1             pyhd3eb1b0_0
pywavelets                1.1.1            py38he774522_2
pywin32                   301                      pypi_0    pypi
pywin32-ctypes            0.2.0                 py38_1000
pywinpty                  0.5.7                    py38_0
pyyaml                    5.4.1                    pypi_0    pypi
pyzmq                     22.1.0                   pypi_0    pypi
qdarkstyle                3.0.2              pyhd3eb1b0_0
qstylizer                 0.1.10             pyhd3eb1b0_0
qt                        5.9.7            vc14h73c81de_0
qtawesome                 1.0.2              pyhd3eb1b0_0
qtconsole                 5.1.0              pyhd3eb1b0_0
qtpy                      1.9.0                      py_0
readlif                   0.6.1                    pypi_0    pypi
regex                     2021.4.4         py38h2bbff1b_0
requests                  2.25.1             pyhd3eb1b0_0
rope                      0.19.0             pyhd3eb1b0_0
rtree                     0.9.7            py38h2eaa2aa_1
scikit-image              0.18.1           py38hf11a4ad_0
scipy                     1.7.0                    pypi_0    pypi
send2trash                1.5.0              pyhd3eb1b0_1
setuptools                52.0.0           py38haa95532_0
sip                       4.19.13          py38ha925a31_0
six                       1.16.0             pyhd3eb1b0_0
snappy                    1.1.8                h33f27b4_0
snowballstemmer           2.1.0              pyhd3eb1b0_0
sortedcontainers          2.3.0              pyhd3eb1b0_0
sphinx                    4.0.2                    pypi_0    pypi
sphinxcontrib-applehelp   1.0.2              pyhd3eb1b0_0
sphinxcontrib-devhelp     1.0.2              pyhd3eb1b0_0
sphinxcontrib-htmlhelp    2.0.0                    pypi_0    pypi
sphinxcontrib-jsmath      1.0.1              pyhd3eb1b0_0
sphinxcontrib-qthelp      1.0.3              pyhd3eb1b0_0
sphinxcontrib-serializinghtml 1.1.5                    pypi_0    pypi
spyder                    5.0.3            py38haa95532_1
spyder-kernels            2.0.3            py38haa95532_0
sqlite                    3.35.4               h2bbff1b_0
tblib                     1.7.0                    pypi_0    pypi
terminado                 0.9.4            py38haa95532_0
testpath                  0.4.4              pyhd3eb1b0_0
text-unidecode            1.3                        py_0
textdistance              4.2.1              pyhd3eb1b0_0
three-merge               0.1.1              pyhd3eb1b0_0
tifffile                  2021.6.14                pypi_0    pypi
tinycss                   0.4             pyhd3eb1b0_1002
tk                        8.6.10               he774522_0
toml                      0.10.2             pyhd3eb1b0_0
toolz                     0.11.1             pyhd3eb1b0_0
tornado                   6.1              py38h2bbff1b_0
tqdm                      4.61.1                   pypi_0    pypi
traitlets                 5.0.5              pyhd3eb1b0_0
typed-ast                 1.4.2            py38h2bbff1b_1
typing-extensions         3.10.0.0                 pypi_0    pypi
typing_extensions         3.7.4.3            pyha847dfd_0
ujson                     4.0.2            py38hd77b12b_0
unidecode                 1.2.0              pyhd3eb1b0_0
urllib3                   1.26.5                   pypi_0    pypi
vc                        14.2                 h21ff451_1
vispy                     0.6.6                    pypi_0    pypi
vs2015_runtime            14.27.29016          h5e58377_2
watchdog                  1.0.2            py38haa95532_1
wcwidth                   0.2.5                      py_0
webencodings              0.5.1                    py38_1
wheel                     0.36.2             pyhd3eb1b0_0
whichcraft                0.6.1              pyhd3eb1b0_0
win_inet_pton             1.1.0            py38haa95532_0
wincertstore              0.2                      py38_0
winpty                    0.4.3                         4
wrapt                     1.12.1           py38he774522_1
xarray                    0.16.2                   pypi_0    pypi
xmlschema                 1.4.1                    pypi_0    pypi
xz                        5.2.5                h62dcd97_0
yaml                      0.2.5                he774522_0
yapf                      0.31.0             pyhd3eb1b0_0
zarr                      2.6.1                    pypi_0    pypi
zeromq                    4.3.3                ha925a31_3
zfp                       0.5.5                hd77b12b_6
zict                      2.0.0                    pypi_0    pypi
zipp                      3.4.1              pyhd3eb1b0_0
zlib                      1.2.11               h62dcd97_4
zstd                      1.4.9                h19a0ad4_0

I'm uploading the image now (~1GB) and will send you the link. Thanks in advance!
```

**Comment by jwindhager** [comment]
Thanks for sending the example image, @jo-mueller! I was able to replicate the issue and released a fix (v0.2.2). The problem was related to undocumented behavior of napari, see https://github.com/napari/napari/issues/2348. With the new version, I was able to open your image. Thank you for using napari-czifile2, I hope you find it helpful for your analyses.

**Comment by jo-mueller** [comment]
Thanks for looking into it! One minor issue, though: In some other data, it seems that the image data's min/max values are not forwarded correctly to the range controls of the contrast slider, so the range has to be set manually through the console. 

Other than that, 0.2.2 works like a charm :)

**Comment by jwindhager** [comment]
> One minor issue, though: In some other data, it seems that the image data's min/max values are not forwarded correctly to the range controls of the contrast slider, so the range has to be set manually through the console.

Huh, that's interesting. napari-czifile2 does not explicitly set the contrast limits, but merely calls napari's [add_image](https://napari.org/api/stable/napari.Viewer.html#napari.Viewer.add_image) function, which is supposed to set the contrast limits to the min/max of the passed image data. If the issue persists, I'd recommend to file an issue with napari directly, they usually respond very quickly.

**Comment by jwindhager** [comment]
Ah, seems like this was reported on napari's side already: https://github.com/napari/napari/issues/2888. I will release a 0.2.3 later today that explicitly sets the `contrast_range`.

**Comment by jwindhager** [comment]
Would you mind sharing the command you use to manually set the range? Just to make sure to be on the same page re contrast ranges.

**Comment by jo-mueller** [comment]
Sorry for the late reply.

In napari, I use these commands:
```
my_layer = viewer.layers['my_layer_name']
my_layer.contrast_limits = [x0, x1]
```

with x0 and x1 being the new range limits.

**Comment by jwindhager** [comment]
Thanks! Released v0.2.4 in the meantime, which - for now - explicitly sets the contrast_limits as described by @sofroniewn here: https://github.com/napari/napari/issues/2888#issuecomment-868939861
